src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -249,17 +252,44 @@ public virtual TestResult Invoke(object?[]? arguments)
         // See https://github.com/microsoft/testfx/issues/4331
         if (expectedExceptions.Count() > 1)
         {
-            string errorMessage = string.Format(
-                CultureInfo.CurrentCulture,
-                Resource.UTA_MultipleExpectedExceptionsOnTestMethod,
-                Parent.ClassType.FullName,
-                TestMethod.Name);
-            throw new TypeInspectionException(errorMessage);
+            ThrowMultipleAttributesException(nameof(ExpectedExceptionBaseAttribute));
         }
 
         return expectedExceptions.FirstOrDefault();
     }
 
+    /// <summary>
+    /// Gets the number of retries this test method should make in case of failure.
+    /// </summary>
+    /// <returns>
+    /// The number of retries, which is always greater than or equal to 1.
+    /// If RetryAttribute is not present, returns 1.
+    /// </returns>
+    private RetryBaseAttribute? GetRetryAttribute()
+    {
+        IEnumerable<RetryBaseAttribute> attributes = ReflectHelper.Instance.GetDerivedAttributes<RetryBaseAttribute>(TestMethod, inherit: true);
+        if (attributes.Count() > 1)
```

Comment:
Not a problem as we don't expect to have many attributes but I would usually recommend to avoid `Count` when you just want to check more than 1.